client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -362,7 +354,7 @@ public void appendSetup(AppendSetup appendSetup) {
                                              .map(entry -> new Append(segmentName, writerId, entry.getKey(),
                                                                       Unpooled.wrappedBuffer(entry.getValue()
                                                                                                   .getData()),
-                                                                      entry.getValue().getExpectedOffset()))
+                                                                      null))
```

Comment:
ok, this is null because all conditional writes are done via the new `ConditionalOutputStream` and we need not specify the `expectedSegmentOffset` inside Append.

Reply:
ConditionalAppend gets deserialized on the server side as an Append object, because it is simpler to keep the server side as a single code path.